### PR TITLE
[Merged by Bors] - feat: port Analysis.NormedSpace.AffineIsometry

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
There are still many linting errors here.

For the first pass, lets ignore the ones of the form:
```
simplify fails on left-hand side:
tactic 'simp' failed, nested error:
(deterministic) timeout at 'whnf', maximum number of heartbeats (200000) has been reached (use 'set_option maxHeartbeats <num>' to set the limit)
```
which I think are caused by the simp linter running without etaExperiment. We may need a mechanism to enable this.

However there are many other linting errors to, which should be fixed first, then we'll see where things lie.

**Comment by mattrobball** [comment]
Should we just run `fail_if_no_progress (simp)` on the remaining 37 `simpNF` errors and `nolint` them if it errors out?

**Comment by mattrobball** [comment]
Apparently `simp` makes progress on 29 declarations.

**Comment by ChrisHughes24** [comment]
> There are still many linting errors here.
> 
> For the first pass, lets ignore the ones of the form:
> 
> ```
> simplify fails on left-hand side:
> tactic 'simp' failed, nested error:
> (deterministic) timeout at 'whnf', maximum number of heartbeats (200000) has been reached (use 'set_option maxHeartbeats <num>' to set the limit)
> ```
> 
> which I think are caused by the simp linter running without etaExperiment. We may need a mechanism to enable this.
> 
> However there are many other linting errors to, which should be fixed first, then we'll see where things lie.

This was all fixed by using `FunLike` and `EquivLike` instead of `CoeFun`.

File: Mathlib/Analysis/NormedSpace/AffineIsometry.lean [hunk 1]
```diff
@@ -0,0 +1,933 @@
...
+
+variable (e : V ≃ₗᵢ[𝕜] V₂)
+
+/-- Reinterpret a linear isometry equiv as an affine isometry equiv. -/
+def toAffineIsometryEquiv : V ≃ᵃⁱ[𝕜] V₂ :=
+  { e.toLinearEquiv.toAffineEquiv with norm_map := e.norm_map }
+#align linear_isometry_equiv.to_affine_isometry_equiv LinearIsometryEquiv.toAffineIsometryEquiv
+
+-- @[simp] -- Porting note: simp-normal form is `coe_toAffineIsometryEquiv'`
+theorem coe_toAffineIsometryEquiv : ⇑(e.toAffineIsometryEquiv : V ≃ᵃⁱ[𝕜] V₂) = e :=
```
Comment (dupuisf) [inline Mathlib/Analysis/NormedSpace/AffineIsometry.lean:409]: What does this porting note mean?

Comment (mattrobball) [reply Mathlib/Analysis/NormedSpace/AffineIsometry.lean:409]: That should be deleted. `coe_toAffineIsometryEquiv’` was provable by `simp` and removed

**Comment by mattrobball** [comment]
Went back through the porting notes concerning the `simpNF` linter and cleaned them up with the changes

File: Mathlib/Analysis/NormedSpace/AffineIsometry.lean (hunk 1 above)
Comment (mattrobball) [reply Mathlib/Analysis/NormedSpace/AffineIsometry.lean]: Done.

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors r+

**Pull request closed without merging**